app/products/calls/state/current_call.ts

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {useEffect, useState} from 'react';
+import {BehaviorSubject} from 'rxjs';
+
+import {CurrentCall} from '@app/products/calls/types/calls';
+
+const currentCallSubject = new BehaviorSubject<CurrentCall | null>(null);
+
+// Used internally, only exported for tests (not exported from the module index).
+export const setCurrentCall = (currentCall: CurrentCall | null) => {
+    currentCallSubject.next(currentCall);
+};
+
+export const getCurrentCall = () => {
+    return currentCallSubject.value;
+};
+
+export const observeCurrentCall = () => {
+    return currentCallSubject.asObservable();
+};
+
+// Used internally (for now). Only exported for tests (not exported from the module index).
+export const useCurrentCall = () => {
```

Comment:
I wonder if it is worth to export it in a `internal` export, so it is clear that it is not meant to be used outside of the package. 0/5